cache/ttlotel.go

```diff
@@ -0,0 +1,49 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+var tracer = otel.Tracer("cache")
+
+const (
+	ttlCacheFoundAttribute = "evergreen.cache.ttl.%s.found"
+)
+
+// newTTLOtelCache wraps a cache and adds OpenTelemetry tracing to it.
+func newTTLOtelCache[T any](cache TTLCache[T]) TTLCache[T] {
+	return &otelTTLCache[T]{cache: cache}
+}
+
+type otelTTLCache[T any] struct {
+	cache TTLCache[T]
+}
+
+func (c *otelTTLCache[T]) Get(ctx context.Context, id string, minimumLifetime time.Duration) (T, bool) {
+	ctx, span := tracer.Start(ctx, "cache.Get")
+	defer span.End()
+
+	value, ok := c.cache.Get(ctx, id, minimumLifetime)
+
+	span.SetAttributes(
+		attribute.Bool(fmt.Sprintf(ttlCacheFoundAttribute, c.name()), ok),
```

Comment:
Since `name()` isn't used anywhere but here, would it be easier to just pass in the name of the cache to `newTTLOtelCache`? It avoids creating an interface that has unexported method that's used for one specific purpose.

Reply:
Yes much easier. It also separates the concerns (why does an in memory cache need to worry about a name?)